datatypes/src/primitives/coordinate.rs

```diff
@@ -392,4 +412,48 @@ mod test {
             2.0_f64.sqrt()
         );
     }
+
+    #[test]
+    fn partial_eq() {
+        let c1 = Coordinate2D { x: 4., y: 8. };
+        let c2 = Coordinate2D { x: 4., y: 8. };
+        assert_eq!(c1, c2);
+    }
+
+    #[test]
+    fn partial_eq_nan() {
+        let c1 = Coordinate2D { x: f64::NAN, y: 8. };
+        let c2 = Coordinate2D { x: f64::NAN, y: 8. };
```

Comment:
I'm not sure if we should allow NAN for coordinates.

Reply:
377a1adcaf72e31a8346ad2a681033d2d21dfc02